### PR TITLE
Issue 176 alt

### PR DESCRIPTION
**Pull request by jrincayc**
Possible fixes for #176

### CODE REVIEW

File: xgraphics.h [hunk 1]
```diff
@@ -22,15 +22,21 @@ void logofill(void);
...
+    if (!have_x) { \
+	err_logo(BAD_GRAPH_INIT,NIL); \
+	return ret; \
+    } \
+}
+#define checkXnoreturn { \
     if (have_x < 0) real_window_init(); \
     if (!have_x) { \
 	err_logo(BAD_GRAPH_INIT,NIL); \
-	return; \
```
Comment (jrincayc) [inline xgraphics.h:29]: I wonder if this is going to cause an error to be missed somewhere?

Comment (jrincayc) [reply xgraphics.h:29]: On the other hand, it possibly needs to return an error code to actually have this noticed?

File: term.c [hunk 2]
```diff
@@ -95,8 +95,8 @@ extern char **environ, *tgoto(), *tgetstr();
 
 char *termcap_ptr;
 
-int termcap_putter(char ch) {
-    *termcap_ptr++ = ch;
+int termcap_putter(int ch) {
+    *termcap_ptr++ = (char)ch;
```
Comment (jrincayc) [inline term.c:99]: Should we be checking for non-char values here?

**Comment by jrincayc** [comment]
I am tempted to add some warning comments and call this good.  It is not code that is used by the main wxWidgets version, so I am less worried about breaking something here. (And the regular regression tests were able to run in x11 mode, and I checked that I could do my smoke test on it.)

**Comment by dmalec** [comment]
Reviewed changes and approve. This looks like a pragmatic approach to addressing the issue.

**Pull request merged**